clang/lib/Frontend/CompilerInstance.cpp

```diff
@@ -105,6 +105,15 @@ void CompilerInstance::setTarget(TargetInfo *Value) { Target = Value; }
 void CompilerInstance::setAuxTarget(TargetInfo *Value) { AuxTarget = Value; }
 
 bool CompilerInstance::createTarget() {
+  if (getLangOpts().DecimalFloatingPoint) {
+    if (getInvocation().getTargetOpts().CPU == "x86" ||
+        getInvocation().getTargetOpts().CPU == "x86-64")
+      getInvocation().getTargetOpts().DFPEncoding =
+          TargetOptions::BinaryEncoding::BID;
+    else
```

Comment:
I don't know that this logic is correct for most targets.

libdfp only mentions x86(-64), PPC and s390 in its documentation (see https://github.com/libdfp/libdfp), but helpfully, the most recent commit suggests that aarch64 is going to go with BID.

The AArch64 suggests that it's using BID: https://github.com/ARM-software/abi-aa/blob/main/aapcs64/aapcs64.rst#decimal-floating-point (there's nothing corresponding I could find for 32-bit arm).

Reply:
It isn't right. I suspect Zahira derived this from similar code I added to the `LinuxTargetInfo` constructor in `clang/lib/Basic/Targets/OSTargets.h`. I'm still working on code to enable proper target dependent enablement. I've been taking forever to get that done.

I think our plan for prior to upstreaming should be to focus on just Intel support (perhaps IBM too given Ariel's involvement) and then let maintainers for other architectures enable support for other targets following appropriate testing after our changes are accepted upstream.

Reply:
Again the change you are suggesting implies that the encoding info is in TargetInfo but I don't think we can do that there. See comments above.